### PR TITLE
Various code logic corrections (without API changes).

### PR DESCRIPTION
**Pull request by Hybris95**
### Description of Change ###

Corrections after passing a PVS-Studio on the Project
No tests because no test environment yet just newly forking the project.

Some corrections may be implemented later (View.cs & LayoutAlignmentExtensions.cs).
Tests needs to be done also.

### PR Checklist ###

- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [ ] Changes adhere to coding standard
- [ ] Consolidate commits as makes sense

### CODE REVIEW

**Comment by Hybris95** [comment]
Last PR : https://github.com/xamarin/Xamarin.Forms/pull/194

File: Xamarin.Forms.Core/ItemsView.cs [hunk 1]
```diff
@@ -88,5 +89,5 @@ static bool ValidateItemTemplate(BindableObject b, object v)
 
 			return !(lv.CachingStrategy == ListViewCachingStrategy.RetainElement && lv.ItemTemplate is DataTemplateSelector);
 		}
-	}
+    }
```
Comment (jassmith) [inline Xamarin.Forms.Core/ItemsView.cs]: Random whitespace change?

File: Xamarin.Forms.Core/LayoutAlignmentExtensions.cs [hunk 2]
```diff
@@ -14,8 +14,9 @@ public static double ToDouble(this LayoutAlignment align)
 					return 0.5;
 				case LayoutAlignment.End:
 					return 1;
-			}
-			throw new ArgumentOutOfRangeException("align");
+                default:// TODO - Maybe there is a specific value for LayoutAlignment.Fill ?
```
Comment (jassmith) [inline Xamarin.Forms.Core/LayoutAlignmentExtensions.cs]: Incorrect idnentation

File: Xamarin.Forms.Core/MultiPage.cs [hunk 3]
```diff
@@ -355,5 +355,5 @@ void UpdateCurrentPage()
 			else if (SelectedItem is T)
 				CurrentPage = (T)SelectedItem;
 		}
-	}
+    }
```
Comment (jassmith) [inline Xamarin.Forms.Core/MultiPage.cs]: Same

File: Xamarin.Forms.Core/SplitOrderedList.cs [hunk 4]
```diff
@@ -455,7 +455,7 @@ public void EnterReadLock()
 					while ((_rwlock & (RwWrite | RwWait)) > 0)
 						sw.SpinOnce();
 
-					if ((Interlocked.Add(ref _rwlock, RwRead) & (RwWait | RwWait)) == 0)
+					if ((Interlocked.Add(ref _rwlock, RwRead) & (RwWrite | RwWait)) == 0)
```
Comment (jassmith) [inline Xamarin.Forms.Core/SplitOrderedList.cs]: +1

File: Xamarin.Forms.Core/View.cs [hunk 5]
```diff
@@ -49,7 +49,9 @@ protected internal View()
 						foreach (IElement item in _gestureRecognizers.OfType<IElement>())
 							item.Parent = this;
 						break;
-				}
+                    case NotifyCollectionChangedAction.Move:
```
Comment (jassmith) [inline Xamarin.Forms.Core/View.cs]: whitespace again

Comment (jassmith) [inline Xamarin.Forms.Core/View.cs]: further we are probably better off tust marking Reset as the Default here

Comment (jassmith) [inline Xamarin.Forms.Core/View.cs]: Also in the case of move this can just be a blank no-op

File: Xamarin.Forms.Core/TemplatedItemsList.cs [hunk 6]
```diff
@@ -1302,6 +1302,9 @@ void OnItemsListCollectionChanged(object sender, NotifyCollectionChangedEventArg
 					case NotifyCollectionChangedAction.Replace:
 						e = new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Replace, ConvertItems(e.NewItems), ConvertItems(e.OldItems), e.OldStartingIndex);
 						break;
+                    case NotifyCollectionChangedAction.Reset:
```
Comment (jassmith) [inline Xamarin.Forms.Core/TemplatedItemsList.cs]: more whitespace

File: Xamarin.Forms.Core/AbsoluteLayout.cs [hunk 7]
```diff
@@ -239,28 +239,25 @@ static Rectangle ComputeLayoutForRegion(View view, Size region)
...
 			}
 			else if (!heightIsProportional && bounds.Height == AutoSize)
 			{
 				// Only height is auto
 				sizeRequest = view.Measure(result.Width, region.Height, MeasureFlags.IncludeMargins);
 				result.Height = sizeRequest.Request.Height;
-			}
-
-			if (xIsProportional)
+            }
```
Comment (jassmith) [inline Xamarin.Forms.Core/AbsoluteLayout.cs]: whitespace :(

**Comment by jassmith** [comment]
Changes to absolute layout break unit tests

File: Xamarin.Forms.Core/View.cs (hunk 5 above)
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/View.cs]: `GestureRecognizers` is exposed as a `IList`, so a `Move` is very unlikely to happen.

File: Xamarin.Forms.Core/AbsoluteLayout.cs [hunk 8]
```diff
@@ -239,28 +239,25 @@ static Rectangle ComputeLayoutForRegion(View view, Size region)
...
+            else
+            {
+                // Only width is auto
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/AbsoluteLayout.cs]: looks like we're loosing a case here when `!heightIsProportional`

**Comment by rmarinho** [comment]
Please re-open when addressing comments, coding style, and unit tests failures.

**Comment by Hybris95** [comment]
Please do it yourself, I have no interest doing so. Fix yourself the bugs I pointed out if you wish to.

**Pull request closed without merging**